### DISCUSSION TITLE
Migration from unencrypted sqlite database to sqlcypher?

### DISCUSSION THREAD

**Discussion by Vorkytaka** [simolus3/drift#2346] [Q&A] [answered]
Case: have an app that use `sqlite3_flutter_libs`, but now we need to encrypt our database, so, as i see, we can use `sqlcipher_flutter_libs`. Is there any way to migrate from one to another and don't loose our data?

As i think we can create new encrypted database, migrate all data from old one and remove it.
Is there any example or guide how we can migrate from unencrypted database to encrypted one?

**Comment by simolus3** [comment, marked as answer]
I recommend taking a look at [`sqlcipher_export`](https://www.zetetic.net/sqlcipher/sqlcipher-api/index.html#sqlcipher_export) for this.

A good way to do this would be to use different file names for the old (unencrypted) and the new database. Then you can put this logic into the opening logic in `LazyDatabase`. Basically, you need to create and export the new encrypted database if the unencrypted database exists.
